examples/change_attribute/change_attitude.py

```diff
@@ -0,0 +1,160 @@
+    # Copter should arm in GUIDED_NOGPS mode
+    vehicle.mode = VehicleMode("GUIDED_NOGPS")
+    vehicle.armed = True
+
+    while not vehicle.armed:
+        print " Waiting for arming..."
+        time.sleep(1)
+
+    print "Taking off!"
+
+    thrust = DEFAULT_TAKEOFF_THRUST
+    while True:
+        current_altitude = vehicle.location.global_relative_frame.alt
+        print " Altitude: ", current_altitude
+        if current_altitude >= aTargetAltitude*0.95: # Trigger just below target alt.
+            print "Reached target altitude"
+            break
+        elif current_altitude >= aTargetAltitude*0.6:
+            thrust = SMOOTH_TAKEOFF_THRUST
+        set_attitude(thrust = thrust)
+        time.sleep(0.2)
+
+
+def set_attitude(roll_angle = 0.0, pitch_angle = 0.0, yaw_rate = 0.0, thrust = 0.5, duration = 0):
+    """
+    Note that from AC3.3 the message should be re-sent every second (after about 3 seconds
+    with no message the velocity will drop back to zero). In AC3.2.1 and earlier the specified
+    velocity persists until it is canceled. The code below should work on either version
+    (sending the message multiple times does not cause problems).
+    """
+    
+    """
+    The roll and pitch rate cannot be controllbed with rate in radian in AC3.4.4 or earlier,
+    so you must use quaternion to control the pitch and roll for those vehicles.
+    """
+    
+    # Thrust >  0.5: Ascend
+    # Thrust == 0.5: Hold the altitude
+    # Thrust <  0.5: Descend
+    msg = vehicle.message_factory.set_attitude_target_encode(
+                                                             0,
+                                                             0,                                         #target system
+                                                             0,                                         #target component
+                                                             0b00000000,                                #type mask: bit 1 is LSB
+                                                             to_quaternion(roll_angle, pitch_angle),    #q
+                                                             0,                                         #body roll rate in radian
+                                                             0,                                         #body pitch rate in radian
+                                                             math.radians(yaw_rate),                    #body yaw rate in radian
+                                                             thrust)                                    #thrust
+    vehicle.send_mavlink(msg)
+                                                             
+    if duration != 0:
+        # Divide the duration into the frational and integer parts
+        modf = math.modf(duration)
+        
+        # Sleep for the fractional part
+        time.sleep(modf[0])
+        
+        # Send command to vehicle on 1 Hz cycle
```

Comment:
Although the timeout for the velocity control is 3s, the timeout for angle control in GUIDED is only 1s. Given that, it might be cleaner to increase this rate to, say, 2Hz or so (and mention this timeout in a comment) to make sure to never accidentally trigger a timeout even briefly.